app/src/main/java/com/nervousfish/nervousfish/data_objects/Profile.java

```diff
@@ -1,57 +1,69 @@
 package com.nervousfish.nervousfish.data_objects;
 
+import com.nervousfish.nervousfish.ConstantKeywords;
+
+import java.io.InvalidObjectException;
+import java.io.ObjectInputStream;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+
 /**
  * An Profile POJO to store a name, public key and private key.
  */
-public class Profile {
+public class Profile implements Serializable {
 
-    private final String name;
-    private final KeyPair keyPair;
+    private static final long serialVersionUID = -4715364587956219157L;
+    private final Contact contact;
+    private final List<KeyPair> keyPairs;
 
     /**
      * The constructor for the {@link Profile} class.
      *
-     * @param name the name belonging to the Profile
-     * @param keyPair the public/private key-pair
+     * @param contact The contact belonging to the user.
+     * @param keyPairs the public/private key-pairs of the user.
      */
-    public Profile(final String name, final KeyPair keyPair) {
-        this.name = name;
-        this.keyPair = keyPair;
+    public Profile(final Contact contact, final List<KeyPair> keyPairs) {
+        this.contact = contact;
+        this.keyPairs = keyPairs;
     }
 
     /**
-     * Get the name of the {@link Profile}.
-     *
-     * @return The name.
+     * Adds a new keyPair to the profile
+     * @param keyPair the keyPair to add.
      */
-    public String getName() {
-        return this.name;
+    public void addKeyPair(final KeyPair keyPair) {
+        keyPairs.add(keyPair);
+        contact.getKeys().add(keyPair.getPublicKey());
     }
 
     /**
-     * Get the {@link KeyPair} of the {@link Profile}.
-     *
-     * @return The {@link KeyPair}.
+     * Returns the contact Object of the user
+     * @return - The Contact POJO.
      */
-    public KeyPair getKeyPair() { return this.keyPair; }
+    public Contact getContact() {
+        return contact;
+    }
 
     /**
-     * Get the public {@link IKey} of the {@link Profile}.
-     *
-     * @return The public {@link IKey}.
+     * Returns the Keypairs of the user.
+     * @return The list of keypairs of the user.
      */
-    public IKey getPublicKey() {
-        return this.keyPair.getPublicKey();
+    public List<KeyPair> getKeyPairs() {
+        return keyPairs;
     }
 
     /**
-     * Get the private {@link IKey} of the {@link Profile}.
-     *
-     * @return The private {@link IKey}.
+     * Returns the (first) public key of the profile.
+     * @return the (first) public key.
      */
-    public IKey getPrivateKey() {
-        return this.keyPair.getPrivateKey();
-    }
+    public IKey getPublicKey() { return keyPairs.get(0).getPublicKey(); }
```

Comment:
Can you add a TODO here that we should let the user choose which keypair is currently "active"